opal/core/patient_lists.py

```diff
@@ -119,6 +119,69 @@ def get_template_prefixes(self):
         return possible
 
 
+"""
+Sometimes we group lists for display purposes.
+"""
+class TabbedPatientListGroup(discoverable.DiscoverableFeature):
+    """
+    Groups of Patient Lists to display as tabs at the top of
+    any list in the group
+    """
+    module_name   = 'patient_lists'
+    member_lists  = []
+
+    @classmethod
+    def for_list(klass, patient_list):
+        """
+        Returns the group for a given PatientList.
+        Raises ValueError if not passed a PatientList
+        """
+        msg = 'TabbedPatientListGroup.for_list must be passed a PatientList'
+        try:
+            if not issubclass(patient_list, PatientList):
+                raise ValueError(msg)
+        except TypeError:
+                raise ValueError(msg)
+
+        for group in klass.list():
+            if patient_list in group.get_member_lists():
+                return group
+
+    @classmethod
+    def get_member_lists(klass):
+        """
+        A hook for dynamically customising the members of this list group.
+
+        Returns an iterable of PatientLists
+        Defaults to the `.member_lists` property
+        """
+        for l in klass.member_lists:
+            yield l
+
+    @classmethod
+    def get_member_lists_for_user(klass, user):
+        """
+        Returns an iterable of the visible member lists for a given USER
+        """
+        for l in klass.get_member_lists():
+            if l.visible_to(user):
+                yield l
+
+    @classmethod
+    def visible_to(klass, user):
+        """
+        Predicate function to determine whether this list is meaningfully
+        visible to this USER
+        """
+        if len(list(klass.get_member_lists_for_user(user))) > 1:
```

Comment:
Don't need in terms of "there's an alternative syntax" ? 

You could spell it like this I guess....
```python 
return bool(list(klass.get_member_lists_for_user(user)))
```

... my reckons would be that the conceptual criteria we're using for visibility is explicitly the length of the thing, so doing it explicitly gives you better readability.


TBH the thing I'd question about this would be if 1 is a less sensible default than 2 here - e.g. is a tabbed list group of two things useful enough to clutter the ui